provider/src/test/kotlin/org/gradle/kotlin/dsl/GroovyInteroperabilityTest.kt

```diff
@@ -107,6 +112,7 @@ class GroovyInteroperabilityTest {
         val builderResult = delegate.withGroovyBuilder {
             val invokeResult = "withKeywordArguments"("string" to "42", "int" to 42)
             assertThat(invokeResult, sameInstance(expectedInvokeResult))
+            assertThat(this.delegate, sameInstance<Any>(delegate))
```

Comment:
❓ Aren't `this.delegate` and `delegate` the same reference?

Reply:
They are not. `delegate` refers to the local variable. The rule is that local variables have precedence over implicit receiver members. But declaring an explaining variable in the outer scope makes sense and I'll do that.

Reply:
> The rule is that local variables have precedence over implicit receiver members.

I keep forgetting this, not obvious to me, I need to stick this in my head

> But declaring an explaining variable in the outer scope makes sense and I'll do that.

👍

Reply:
Yeah, I remember this rule as the only disappointment I had with Kotlin so far.

Reply:
@eskatos Done.